handlers/metrics.go

```diff
@@ -80,6 +84,37 @@ func extractMetricsQueryParams(r *http.Request, q *prometheus.MetricsQuery) erro
 		q.ByLabelsOut = lblsout
 	}
 
+	// Get namespace info.
+	namespace, err := k8s.GetNamespace(q.Namespace)
+	if err != nil {
+		return err
+	}
+	// If needed, adjust interval -- Make sure query won't fetch data before the namespace creation
+	intervalStartTime, err := util.GetStartTimeForRateInterval(q.End, q.RateInterval)
+	if err != nil {
+		return err
+	}
+	if intervalStartTime.Before(namespace.CreationTimestamp.Time) {
+		q.RateInterval = fmt.Sprintf("%vs", int(q.End.Sub(namespace.CreationTimestamp.Time).Seconds()))
+		intervalStartTime = namespace.CreationTimestamp.Time
+		log.Debugf("[extractMetricsQueryParams] Interval set to: %v", q.RateInterval)
+	}
+	// If needed, adjust query start time (bound to namespace creation time)
+	log.Debugf("[extractMetricsQueryParams] Requested query start time: %v", q.Start)
+	intervalDuration := q.End.Sub(intervalStartTime)
+	allowedStart := namespace.CreationTimestamp.Time.Add(intervalDuration)
```

Comment:
I have a doubt here, why allowedStart isn't just namespace.CreationTimestamp.Time ?
In other words, why not just:

```
if q.Start.Before(namespace.CreationTimestamp.Time) {
   q.Start = namespace.CreationTimestamp.Time
```

I must be missing something?

Reply:
Because of the _interval_. We don't want to query data logged before the namespace creation date. Setting `q.Start = namespace.CreationTimestamp.Time` may lead to prometheus evaluate the query at `namespace.CreationTimestamp.Time` and using data for an `interval` past that date. This is my reasoning.

From reading the docs, I understood that `q.Start` and `q.End` only sets a range to evaluate the query in steps of length `q.Step`. This doesn't really set "boundaries" when evaluating a query and fetching data. But, I may be misunderstanding how prometheus evaluates queries. So, If you think that `q.Start = namespace.CreationTimestamp.Time` is enough, I'll change.

Reply:
Ok I see the problem.

But it creates another problem, it also means that when you create a new namespace and immediately start to look at it in kiali, you will always miss the first datapoints (or even get errors), until the delay of "rate interval" is passed, right? Which is probably ok for a 1m rate interval, but could be annoying for something bigger like 30m.

Reply:
Probably, that case is already covered by the combination of adjusting the rate interval (some lines before this one) and adjusting the start date. The "rate interval" should be adjusted to the time of live of the namespace and here, the the start of the range should be set so that `q.Start == q.End`. And that will, at least, return some data.

I don't know what's better. So, if you think that is better to use irate, I'll change this.